### ISSUE TITLE
PEP 541 Request: pdbfixer

### ISSUE THREAD

**Issue by peastman** [pypi/support#5093]
### Project to be claimed

`pdbfixer`: https://pypi.org/project/pdbfixer

### Your PyPI username

`peastman`: https://pypi.org/user/peastman

### Reasons for the request

I am trying to post a wheel for PDBFixer (https://github.com/openmm/pdbfixer).  I am the lead developer of this project.  The upload fails with the error

```
ERROR    HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/                                                                                                
         The name 'pdbfixer' isn't allowed. See https://pypi.org/help/#project-name for more information.                                                               
```

I read the information at the cited URL, but it isn't clear to me what the problem is.  There is no existing package at https://pypi.org/project/pdbfixer.  It is not a package from the standard library, or any name that ought to be blocked.

### Maintenance or replacement?

None

### Source code repositories URLs

https://github.com/openmm/pdbfixer

### Contact and additional research

So far as I can tell, there is no current package with the same name.  I do not know why the upload fails.

### Code of Conduct

- [x] I agree to follow the PSF Code of Conduct

**Comment by peastman** [comment]
Can anyone help with this?  As far as I can tell, there is no existing package with that name.  But it tells me the name isn't allowed with no explanation of why.  I'm at a loss for what to do about it.

**Comment by mattwthompson** [comment]
The project https://pypi.org/project/black-pdbfixer/ of similar name was uploaded by somebody other than @peastman, tagged with the wrong license (GPLv3, the project is [actually MIT](https://github.com/openmm/pdbfixer/blob/master/LICENSE)) and with dead and incorrect links for its homepage/bugtracker. I don't know how PyPI checks its uploads against existing projects, but something like hashing the tarball might be the cause of this error (or at least that seems more likely than `pdbfixer` being a forbidden name).

There is an email listed at that project which I can't find associated with any GitHub username: julian.m.kleber@gmail.com

But fighting for that project name shouldn't be needed here, Peter is not trying to upload to `black-pdbfixer`

**Comment by ex-rzr** [comment]
@cap-jmk
Hi! Are you an uploader of https://pypi.org/project/black-pdbfixer/ ?
If you are, could you remove it to make possible to upload an official version?

**Comment by Thespi-Brain** [comment]
Hi @peastman,

The project name that you wanted is unfortunately a name that was marked as prohibited. A name can be deemed prohibited if it had posed a security risk before such as being associated with malware or any other malicious intent. We are working on determining future perceived risk to decide which prohibited names can be released and which ones will continue to stay in that bucket. Thanks for your patience.

**Comment by peastman** [comment]
Thanks for the reply.  Is there anything I can do to help out?

This isn't a new project.  We've been developing and distributing it under that name for over ten years (https://github.com/openmm/pdbfixer).  We've also had it as a package with that name on conda-forge for over four years (https://anaconda.org/conda-forge/pdbfixer).  If someone posted a malicious package on PyPI with that name, it was undoubtedly to catch people who were looking for the real PDBFixer.

**Comment by peastman** [comment]
Is there anything that can be done to resolve this?  We're trying to migrate the whole OpenMM ecosystem from conda to PyPI.  PDBFixer is one of the core packages that other packages depend on.  Our migration has been blocked for six months by this issue.  We're happy to help however we can, but someone needs to tell us what to do!

**Comment by mikemhenry** [comment]
@Thespi-Brain Hey any updates on this?

**Comment by mattwthompson** [comment]
Perhaps @cap-jmk-real is the author of https://pypi.org/project/black-pdbfixer/?

**Comment by cap-jmk-real** [comment]
> Perhaps [@cap-jmk-real](https://github.com/cap-jmk-real) is the author of https://pypi.org/project/black-pdbfixer/?

Yes, I forked that for a project a couple of months ago. You can use it or get inspired about how to make your version pip-installable. I can also transfer the ownership, as I found another way on how to do it that fits my workflow better. 

Edit: After going through the whole problem, I can offer to remove the package from PyPi, as I would fully support migrating OpenMM to PyPi (is what I needed to do as well). My phone was stolen back then, so I could not use my old GitHub-Account any longer, that's why I did not see your request. 

Please let me know what you want to do: Transfer the ownership or removing the package (However, I do not think the package name is what is causing the error in PyPi). I did choose a completely different name to avoid any clashes with your ecosystem, and I have done similar things of forks before without any of such issues.  I am deeply sorry for the delay, however (it literally also hurt my developments)

**Comment by epretti** [comment]
> However, I do not think the package name is what is causing the error in PyPi

I think you are right; the initial response we got from @Thespi-Brain on January 8, 2025 indicated that `pdbfixer` is on a list of prohibited names (possibly due to a malicious package having been uploaded under that name in the past).  Regardless of how it got there in the first place, I don't think the existence of `black-pdbfixer` is the problem currently, and I don't think we can do anything to be able to use the `pdbfixer` name without help from someone at PyPI.

**Comment by Augustin-Zidek** [comment]
I would also like to voice our support for this. PDBFixer not being on PyPI is preventing AlphaFold 2 from simplifying its setup to be Conda-free. Is there anything we can help with to make this happen?

**Comment by k9cdt** [comment]
I would also express my support for this issue. Many downstream packages for molecular simulations depend on the OpenMM ecosystem, and the absence of this package on PyPi has become the choke point of migrating out from conda. @Thespi-Brain

**Comment by pckroon** [comment]
As a downstream user trying to install pdbfixer I think this is becoming a larger and more serious issue the longer this is delayed/dragged on. There are currently already 2 pdbfixer forks on pypi (https://pypi.org/project/pdbfixer-wheel/ and https://pypi.org/project/pdbfixer-neoralab/). I now need to go through some 30 commits to see where they differ, and whether I can trust them. 
The longer this remains unresolved, the more forks appear, which increases the likelihood of yet another malicious actor abusing the opportunity that is created here.

**Comment by peastman** [comment]
Can anyone associated with PyPI *please* comment on this issue and tell us what we need to do to get the name released?  `pdbfixer` is the name of the conda package.  If we use a different name for the pip package, then anyone who tries to combine conda and pip packages in the same environment will likely end up with two copies of PDBFixer installed and a broken environment.  I really want to avoid that.

It's been well over a year with no official response.  Does anyone even monitor issues here?  If not, is there some other way of reaching someone who can address this?

**Comment by di** [comment]
I've released `pdbfixer` to `peastman`.

**Comment by peastman** [comment]
Thank you!!!!!

**Comment by peastman** [comment]
And for everyone who's been waiting for it: the package is now online! :)
